build.rs

```diff
@@ -12,6 +12,6 @@ fn main() {
         .unwrap();
     println!("cargo:rustc-link-search={}", out.display());
 
-    println!("cargo:rerun-if-changed=build.rs");
+    // Only re-run the build script when memory.x is changed, instead of at every build
```

Comment:
Nice to know that `rerun-if-changed=build.rs` is no longer needed!

"at every build" is not correct.

If `rerun-if-changed` is not present the build script only runs when any file within the Cargo project source (i.e. not in the `target` directory) changes. `cargo build; cargo build` will only run the build script *once*, **unless** the build script is modifying something in the Cargo project source, which is something you should avoid doing exactly because it causes the build script to fire on every `cargo build`. Like the guide says build scripts should put any generated files / output in `OUT_DIR` (which resides within the `target` directory).

TL;DR this should say something like "instead of when any part of the source code changes"

Reply:
You're right, I meant "whenever any of the source code changes" instead of "at every build". For context in this case the build script was using [built](https://crates.io/crates/built) to generate build-time information including the git commit and dirty flag; obviously this needs the build script to actually run every time the source code changes.